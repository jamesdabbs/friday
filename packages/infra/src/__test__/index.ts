import { Testing } from 'cdk8s';

export function synth(construct: any, { name = 'construct', ...params }: Record<string, unknown> = {}) {
  const app = Testing.app();
  const chart = new construct(app, name, params);
  return Testing.synth(chart);
}

export function find(chart: any[], kind: string, name?: string) {
  return chart.find(value => {
    if (value.kind !== kind) { return false }
    if (name && value.metadata?.name !== name) { return false }

    return true
  })
}
