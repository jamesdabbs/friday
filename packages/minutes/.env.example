CONFLUENCE_EMAIL=...
CONFLUENCE_API_TOKEN=...
CONFLUENCE_DOMAIN=...
