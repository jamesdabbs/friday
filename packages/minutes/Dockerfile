FROM node:16-slim AS build
WORKDIR /app

COPY package* /app/
RUN --mount=type=secret,id=npmrc,dst=/app/.npmrc --mount=type=cache,target=~/.npm npm ci

COPY . .

ENV NODE_ENV=development
CMD ["npm", "run", "start"]
