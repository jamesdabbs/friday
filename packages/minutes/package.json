{
  "name": "minutes",
  "version": "0.0.1",
  "description": "",
  "author": "James Dabbs <james.dabbs@gmail.com>",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "start": "nest start",
    "start:dev": "nest-platform dev",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main"
  },
  "dependencies": {
    "@jamesdabbs/nest-platform": "0.0.4",
    "@nestjs/common": "^7.6.15",
    "@nestjs/config": "^0.6.3",
    "@nestjs/core": "^7.6.15",
    "@nestjs/platform-express": "^7.6.15",
    "axios": "^0.21.1",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^6.6.6"
  },
  "devDependencies": {
    "@nestjs/cli": "^7.6.0",
    "@nestjs/schematics": "^7.3.0",
    "@nestjs/testing": "^7.6.15",
    "@types/express": "^4.17.11",
    "@types/jest": "^26.0.22",
    "@types/node": "^14.14.36",
    "@types/supertest": "^2.0.10",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "ts-loader": "^8.0.18",
    "ts-node": "^9.1.1",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.2.3"
  }
}
