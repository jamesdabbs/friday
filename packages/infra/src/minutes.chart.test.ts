import { MinutesChart } from './minutes.chart';
import { synth } from './__test__';

describe(MinutesChart, () => {
  describe('with defaults', () => {
    const chart = synth(MinutesChart)

    it('matches the snapshot', () => {
      expect(chart).toMatchSnapshot()
    })
  })
});
