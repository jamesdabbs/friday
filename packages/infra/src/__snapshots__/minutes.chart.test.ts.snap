// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinutesChart with defaults matches the snapshot 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "minutes",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "construct-construct.nest-minutes.deployment-c80a8646",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "construct-construct.nest-minutes.deployment-c80a8646",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "CONFLUENCE_EMAIL",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_EMAIL",
                      "name": "minutes-secret",
                    },
                  },
                },
                Object {
                  "name": "CONFLUENCE_API_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_API_TOKEN",
                      "name": "minutes-secret",
                    },
                  },
                },
                Object {
                  "name": "CONFLUENCE_DOMAIN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_DOMAIN",
                      "name": "minutes-secret",
                    },
                  },
                },
              ],
              "image": "minutes",
              "imagePullPolicy": "IfNotPresent",
              "livenessProbe": Object {
                "failureThreshold": 3,
                "httpGet": Object {
                  "path": "/health",
                  "port": 3000,
                },
              },
              "name": "minutes",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "minutes",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "cdk8s.deployment": "construct-construct.nest-minutes.deployment-c80a8646",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "minutes.ingress.local",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "minutes.localhost",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "minutes",
                  "servicePort": 80,
                },
              },
            ],
          },
        },
      ],
    },
  },
]
`;
